### DISCUSSION TITLE
Starship module for Mise

### DISCUSSION THREAD

**Discussion by Ajpantuso** [jdx/mise#1554] [Ideas]
A request, https://github.com/starship/starship/issues/5654, was made on the https://github.com/starship/starship project for Mise(rtx) support by way of a new module. I am happy to implement this, but need some help with discovery.

I have already worked with _Starship_ a bit, see the [direnv](https://github.com/starship/starship/blob/2d4063c40ac478c0efe98b27dca3e4299616afa8/src/modules/direnv.rs) module, and could lead the implementation for a _mise_ module. The primary question is what information would you all like to see in the _Starship_ prompt?

### Some Ideas

- Prompt shows if all tool versions in the local mise config are installed and the shims are available on the path
- Prompt shows if `mise doctor` reports healthy or unhealthy
- Prompt shows `mise version` (not sure how relevant that is to users)

If any of you already have some ideas or some custom solutions you would like to share it would be much appreciated, thanks!

**Comment by jdx** [comment]
in general I like the idea of a "healthy/not healthy" flag. That would keep it from getting too noisy. Maybe an option to show the count of active tools + env vars?

actually one thing that might be good is a count of missing tools

**Comment by Ajpantuso** [comment]
First draft PR https://github.com/starship/starship/pull/5747

**Comment by JamBalaya56562** [comment]
This shipped. @Ajpantuso's starship#5747 was merged **2025-04-26**, and starship has had a `mise` module since.

It is **disabled by default**:

```toml
# ~/.config/starship.toml
[mise]
disabled = false
```

What it reports is the healthy/not-healthy flag @jdx asked for in this thread rather than a tool list. From `src/modules/mise.rs`, the `health` variable is filled by running `mise doctor`: success gives `healthy_symbol` (default `healthy`), failure gives `unhealthy_symbol` (default `unhealthy`). Defaults are `format = 'on [$symbol$health]($style) '`, `symbol = 'mise '`, `style = 'bold purple'`, triggered by `detect_files = ['mise.toml', 'mise.local.toml', '.mise.toml', '.mise.local.toml']` and `detect_folders = ['.mise']`.

Two things worth knowing before turning it on:

1. It shells out to **`mise doctor` on every prompt** in a matching directory, which is a much heavier command than the `mise hook-env` that activation runs.
2. The example in starship's own docs is wrong. It shows `[mise] health = 'ready'`, but `health` is a *variable*, not an option — `MiseConfig` (`src/configs/mise.rs`) has no `health` field and is declared `deny_unknown_fields`. The option that changes that text is `healthy_symbol`.

The "count of missing tools" idea from later in the thread did not make it in; the module only reports the binary healthy/unhealthy state.
